src/vivarium_public_health/risks/distributions.py

```diff
@@ -4,21 +4,26 @@
 from risk_distributions import risk_distributions
 
 from vivarium.framework.values import list_combiner, joint_value_post_processor
-from functools import partial
-
+from vivarium_public_health.risks.data_transformations import pivot_categorical
 
 class MissingDataError(Exception):
     pass
 
 
-class EnsembleSimulation(risk_distributions.EnsembleDistribution):
+class EnsembleSimulation:
+    def __init__(self, weights, mean, sd):
+        self._weights = pivot_categorical(weights)
+        self._mean = mean
+        self._sd = sd
 
     def setup(self, builder):
-        builder.components.add_components(self._distributions.values())
+        self.weights = builder.lookup.build_table(self._weights)
+        self.mean = builder.lookup.build_table(self._mean.drop('parameter', axis=1))
+        self.sd = builder.lookup.build_table(self._sd)
 
-    def get_distribution_map(self):
-        dist_map = super().get_distribution_map()
-        return {dist_name: partial(SimulationDistribution, distribution=dist) for dist_name, dist in dist_map.items()}
+    def ppf(self, x):
```

Comment:
oh this is why it's so slow. You're recalculating all the distribution params on each call to this whereas before it happened once. I think you want to do it like we do SimulationDistribution where we pre-calc the params and then put those in a lookup table